internal/app/app.go

```diff
@@ -86,3 +92,27 @@ func display(w io.Writer, summary *parse.GoTestSummary, option Options) {
 	cw.printFailed(packages)
 	cw.summaryTable(packages, option.ShowNoTests, option.SummaryTableOptions)
 }
+
+// displayEvent reads from event channel ec and writes to cw when an event needs
+// to be displayed.
+func displayEvent(cw *consoleWriter, ec <-chan *parse.Event, o Options) {
```

Comment:
> I noticed you'd left a TODO in parse/process.go hinting at passing events back to the caller using a channel. I've taken that approach here, adding displayEvent() and moving the FollowOutput printer there.

Hehe, indeed this is kind of what I originally had in mind when I wrote that comment 😅 

I'm still trying to figure out the best contract with `parse.Process`, but passing the event back on a channel seems sane.

Putting that aside, the reason I punted on `-progress` is b/e I wanted the local vs CI experience to be different.

For non-CI (read local development) I was thinking of using https://github.com/charmbracelet/bubbles#spinner to render a spinner and some periodically updating set of output per package with a test counter that kept going up. I don't think this would work for CI though, esp. ones without a interactive TTY like GitHub Actions.

I also took a stab by keeping things simple and in-lined that output in the parsing loop, https://github.com/mfridman/tparse/pull/95 ..

Reply:
#95 looks good. :+1: The simple solution is probably preferable, I don't love the channel approach. It adds indirection and complexity, and care needs to be taken to not send on a nil channel when progress is disabled.

My 2c: I would avoid the spinner and try to keep CI vs. local differences to a minimum for as long as possible. There's no standard for 'detecting' CI, it always runs on a remote machine, and the last thing a user wants to be doing is troubleshooting tparse. It should be as low-maintenance as possible, with few to no surprises.